src/test/java/se/liu/ida/hefquin/engine/data/impl/VocabularyMappingTest.java

```diff
@@ -101,5 +103,68 @@ public void TranslateTriplePatternTest() {
 		
 		return new Pair<>(testSet, expectedRes);
 	}
+	*/
+	
+	@Test
+	public void TranslateSolutionMappingTest() {
+		final Set<Triple> testSet = new HashSet<>();
+		Node s = NodeFactory.createURI("a");
+		Node p = OWL.equivalentClass.asNode();
+		Node o = NodeFactory.createURI("n");
+		testSet.add(new Triple(s, p ,o));
+		
+		s = NodeFactory.createURI("b");
+		p = OWL.unionOf.asNode();
+		o = NodeFactory.createBlankNode();
+		testSet.add(new Triple(s, p ,o));
+		
+		s = o;
+		p = RDF.first.asNode();
+		o = NodeFactory.createURI("c");
+		testSet.add(new Triple(s, p ,o));
+		
+		p = RDF.rest.asNode();
+		o = NodeFactory.createBlankNode();
+		testSet.add(new Triple(s, p ,o));
+		
+		s = o;
+		p = RDF.first.asNode();
+		o = NodeFactory.createURI("n");
+		testSet.add(new Triple(s, p ,o));
+		
+		p = RDF.rest.asNode();
+		o = RDF.nil.asNode();
+		testSet.add(new Triple(s, p ,o));
+		
+		s = NodeFactory.createURI("c");
+		p = OWL.equivalentClass.asNode();
+		o = NodeFactory.createURI("m");
+		testSet.add(new Triple(s, p ,o));
+		
+		final VocabularyMapping vm = new VocabularyMappingImpl(testSet);
+		
+		final BindingBuilder testBuilder = BindingBuilder.create();
+		testBuilder.add(Var.alloc("v"), NodeFactory.createURI("n"));
+		testBuilder.add(Var.alloc("w"), NodeFactory.createURI("m"));
+		
+		final SolutionMapping testSm = new SolutionMappingImpl(testBuilder.build());		
+		Set<SolutionMapping> translation = vm.translateSolutionMapping(testSm);
```

Comment:
It would be best if the test for translating solution mappings does not rely on the code of vocabulary mapping. They should be tested independently. To do this, please create a set of solution mappings that use a local vocabulary, create a vocabulary mapping, and then translate the set of solution mappings to the global vocabulary.

Reply:
I don't really understand what you mean, should I do the translation, as is done in the function directly in the test rather than using the function in vocabulary mapping? Right now I just create a vocabulary mapping, create a solution mapping and then use the translate solution mapping function to translate it.

Reply:
Oh, I'm so sorry. My IDE was telling me that the `translateSolutionMapping` method didn't exist so I assumed you were calling the triple pattern translation function.